src/index.ts

```diff
@@ -189,18 +210,33 @@ function getterFromOptions(options: PostGraphileOptions) {
   return getter;
 }
 
+function allowUnpersistedOperationsFromOptions(options: PostGraphileOptions, request: IncomingMessage, payload: any): boolean {
+  if (options.allowUnpersistedOperation === undefined || options.allowUnpersistedOperation === null) {
+    return false;
+  }
+  if (typeof options.allowUnpersistedOperation === 'boolean') {
+    return options.allowUnpersistedOperation;
+  }
+  return options.allowUnpersistedOperation(request, payload);
```

Comment:
Slightly simplified, and won't throw error if `options.allowUnpersistedOperation` is a string, object or array:

```suggestion
  const { allowUnpersistedOperation } = options;
  if (typeof allowUnpersistedOperation === 'function') {
    return allowUnpersistedOperation(request, payload);
  }
  return !!allowUnpersistedOperation;
```